src/_pytest/nodes.py

```diff
@@ -425,16 +434,19 @@ def add_report_section(self, when: str, key: str, content: str) -> None:
         if content:
             self._report_sections.append((when, key, content))
 
-    def reportinfo(self):
+    def reportinfo(self) -> Tuple[str, Optional[int], str]:
         return self.fspath, None, ""
 
     @property
-    def location(self):
+    def location(self) -> Tuple[str, Optional[int], str]:
         try:
             return self._location
         except AttributeError:
             location = self.reportinfo()
             fspath = self.session._node_location_to_relpath(location[0])
-            location = (fspath, location[1], str(location[2]))
-            self._location = location
-            return location
+            self._location = (
+                fspath,
+                location[1],
+                str(location[2]),
```

Comment:
`self.reportinfo` type says `location[2]` is an `str`. So either its inaccurate, or the `str()` here is redundant.

Reply:
will harden it with an assert.